File: php/number_guessing_game/number_guessing_game_4.php
<?php
// This is a number guessing game written in PHP. The program generates a random number between 1 and 100.
// The user is prompted to guess the number, and the program provides feedback on whether the guess is too high or too low.
// The game continues until the user guesses the correct number.

$randomNumber = rand(1, 100);
$guess = null;

while ($guess != $randomNumber) {
    echo "Guess the number (between 1 and 100): ";
    $guess = intval(fgets(STDIN));

    if ($guess < $randomNumber) {
        echo "Too low! Try again.\n";
    } elseif ($guess > $randomNumber) {
        echo "Too high! Try again.\n";
    } else {
        echo "Congratulations! You guessed the correct number.\n";
    }
}

echo "Game over.\n";

?>

File: go/number_guessing_game/number_guessing_game_4.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

// Number Guessing Game
// This program generates a random number between 1 and 100 and asks the user to guess it.
// The program provides feedback on whether the guess is too high, too low, or correct.

func main() {
    rand.Seed(time.Now().UnixNano())
    target := rand.Intn(100) + 1
    var guess int

    for {
        fmt.Print("Enter your guess (1-100): ")
        fmt.Scan(&guess)

        if guess < target {
            fmt.Println("Too low!")
        } else if guess > target {
            fmt.Println("Too high!")
        } else {
            fmt.Println("Congratulations! You guessed it!")
            break
        }
    }

}